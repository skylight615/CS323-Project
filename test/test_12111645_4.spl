struct my_struct
{
  int code;
};
int test(int a,int b)
{
    struct my_struct d;
    int f=(2+3/(2-2))*2;
    if(a>=b)
    {
        if(a>b)
        {
            a=1+1;
            b=1+1;
        }
    }
    else{
        a=1;
    }
    if(a>=b)
    {
        if(a>1)
        {
            a=1;
        }
        else
        {
            a=2;
        }
    }
    return a*b;
}
int test(int a,int B){
    while(a<4){
        while(b<3){
            a=a+1;
            b=b+1;
        }
        b=b+1;
    }
}
