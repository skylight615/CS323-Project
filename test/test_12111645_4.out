Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        StructSpecifier (1)
          STRUCT
          ID: my_struct
          LC
          DefList (3)
            Def (3)
              Specifier (3)
                TYPE: int
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: code
              SEMI
          RC
      SEMI
    ExtDefList (5)
      ExtDef (5)
        Specifier (5)
          TYPE: int
        FunDec (5)
          ID: test
          LP
          VarList (5)
            ParamDec (5)
              Specifier (5)
                TYPE: int
              VarDec (5)
                ID: a
            COMMA
            VarList (5)
              ParamDec (5)
                Specifier (5)
                  TYPE: int
                VarDec (5)
                  ID: b
          RP
        CompSt (6)
          LC
          DefList (7)
            Def (7)
              Specifier (7)
                StructSpecifier (7)
                  STRUCT
                  ID: my_struct
              DecList (7)
                Dec (7)
                  VarDec (7)
                    ID: d
              SEMI
            DefList (8)
              Def (8)
                Specifier (8)
                  TYPE: int
                DecList (8)
                  Dec (8)
                    VarDec (8)
                      ID: f
                    ASSIGN
                    Exp (8)
                      Exp (8)
                        LP
                        Exp (8)
                          Exp (8)
                            INT: 2
                          PLUS
                          Exp (8)
                            Exp (8)
                              INT: 3
                            DIV
                            Exp (8)
                              LP
                              Exp (8)
                                Exp (8)
                                  INT: 2
                                MINUS
                                Exp (8)
                                  INT: 2
                              RP
                        RP
                      MUL
                      Exp (8)
                        INT: 2
                SEMI
          StmtList (9)
            Stmt (9)
              IF
              LP
              Exp (9)
                Exp (9)
                  ID: a
                GE
                Exp (9)
                  ID: b
              RP
              Stmt (10)
                CompSt (10)
                  LC
                  StmtList (11)
                    Stmt (11)
                      IF
                      LP
                      Exp (11)
                        Exp (11)
                          ID: a
                        GT
                        Exp (11)
                          ID: b
                      RP
                      Stmt (12)
                        CompSt (12)
                          LC
                          StmtList (13)
                            Stmt (13)
                              Exp (13)
                                Exp (13)
                                  ID: a
                                ASSIGN
                                Exp (13)
                                  Exp (13)
                                    INT: 1
                                  PLUS
                                  Exp (13)
                                    INT: 1
                              SEMI
                            StmtList (14)
                              Stmt (14)
                                Exp (14)
                                  Exp (14)
                                    ID: b
                                  ASSIGN
                                  Exp (14)
                                    Exp (14)
                                      INT: 1
                                    PLUS
                                    Exp (14)
                                      INT: 1
                                SEMI
                          RC
                  RC
              ELSE
              Stmt (17)
                CompSt (17)
                  LC
                  StmtList (18)
                    Stmt (18)
                      Exp (18)
                        Exp (18)
                          ID: a
                        ASSIGN
                        Exp (18)
                          INT: 1
                      SEMI
                  RC
            StmtList (20)
              Stmt (20)
                IF
                LP
                Exp (20)
                  Exp (20)
                    ID: a
                  GE
                  Exp (20)
                    ID: b
                RP
                Stmt (21)
                  CompSt (21)
                    LC
                    StmtList (22)
                      Stmt (22)
                        IF
                        LP
                        Exp (22)
                          Exp (22)
                            ID: a
                          GT
                          Exp (22)
                            INT: 1
                        RP
                        Stmt (23)
                          CompSt (23)
                            LC
                            StmtList (24)
                              Stmt (24)
                                Exp (24)
                                  Exp (24)
                                    ID: a
                                  ASSIGN
                                  Exp (24)
                                    INT: 1
                                SEMI
                            RC
                        ELSE
                        Stmt (27)
                          CompSt (27)
                            LC
                            StmtList (28)
                              Stmt (28)
                                Exp (28)
                                  Exp (28)
                                    ID: a
                                  ASSIGN
                                  Exp (28)
                                    INT: 2
                                SEMI
                            RC
                    RC
              StmtList (31)
                Stmt (31)
                  RETURN
                  Exp (31)
                    Exp (31)
                      ID: a
                    MUL
                    Exp (31)
                      ID: b
                  SEMI
          RC
      ExtDefList (33)
        ExtDef (33)
          Specifier (33)
            TYPE: int
          FunDec (33)
            ID: test
            LP
            VarList (33)
              ParamDec (33)
                Specifier (33)
                  TYPE: int
                VarDec (33)
                  ID: a
              COMMA
              VarList (33)
                ParamDec (33)
                  Specifier (33)
                    TYPE: int
                  VarDec (33)
                    ID: B
            RP
          CompSt (33)
            LC
            StmtList (34)
              Stmt (34)
                WHILE
                LP
                Exp (34)
                  Exp (34)
                    ID: a
                  LT
                  Exp (34)
                    INT: 4
                RP
                Stmt (34)
                  CompSt (34)
                    LC
                    StmtList (35)
                      Stmt (35)
                        WHILE
                        LP
                        Exp (35)
                          Exp (35)
                            ID: b
                          LT
                          Exp (35)
                            INT: 3
                        RP
                        Stmt (35)
                          CompSt (35)
                            LC
                            StmtList (36)
                              Stmt (36)
                                Exp (36)
                                  Exp (36)
                                    ID: a
                                  ASSIGN
                                  Exp (36)
                                    Exp (36)
                                      ID: a
                                    PLUS
                                    Exp (36)
                                      INT: 1
                                SEMI
                              StmtList (37)
                                Stmt (37)
                                  Exp (37)
                                    Exp (37)
                                      ID: b
                                    ASSIGN
                                    Exp (37)
                                      Exp (37)
                                        ID: b
                                      PLUS
                                      Exp (37)
                                        INT: 1
                                  SEMI
                            RC
                      StmtList (39)
                        Stmt (39)
                          Exp (39)
                            Exp (39)
                              ID: b
                            ASSIGN
                            Exp (39)
                              Exp (39)
                                ID: b
                              PLUS
                              Exp (39)
                                INT: 1
                          SEMI
                    RC
            RC
